clang/lib/CIR/CodeGen/CIRGenExprAgg.cpp

```diff
@@ -895,7 +895,7 @@ void AggExprEmitter::VisitLambdaExpr(LambdaExpr *E) {
       ValueDecl *v = capture.getCapturedVar();
       fieldName = v->getName();
       CGF.getCIRGenModule().LambdaFieldToName[*CurField] = fieldName;
-    } else {
+    } else if (!capture.capturesThis()) {
```

Comment:
In this change you are ignoring the field name for "this", why? OG isn't doing name mapping, we do it because it's nicer to read CIR output and map things to source code, as you can probably see in the tests. I'm confused why you deliberately decided not to do it and just ignore it instead - it's a probably a good idea to try to understand what's going on the code that gets touched instead of just copy-n-paste. If you are unsure, ask during the review.

Reply:
Thanks for clarification. Will add it into the map. I was waiting for this comment as I wasn't sure if we want to have `this` to be part of naming map, as it sounded to me we wanted to track variable names only.